lib/tasks/rake_helpers/storage.rb

```diff
@@ -105,16 +105,74 @@ def self.create_dummy_paperclip_files_for(model)
       exit
     end
 
-    ATTACHMENTS[model].each do |name|
-      records = self.models(model).where.not("#{name}_file_name" => nil)
+    ATTACHMENTS[model].each do |attachment|
+      if model.eql?('documents')
+        records = DummyDocument.where.not("#{attachment}_file_name" => nil)
+      else
+        records = self.models(model).where.not("#{attachment}_file_name" => nil)
+      end
 
-      bar = self.progress_bar title: name, total: records.count
+      bar = self.progress_bar title: attachment, total: records.count
 
       records.each do |record|
         bar.increment
-        filename = File.absolute_path(record.send(name).path)
-        FileUtils.mkdir_p File.dirname(filename)
-        File.open(filename, 'wb') { |file| file.write(SecureRandom.random_bytes(record.send("#{name}_file_size"))) }
+
+        filename = if paperclip_storage.eql?(:s3)
+                     File.join('tmp', record.send(attachment).path)
+                   else
+                     File.absolute_path(record.send(attachment).path)
+                   end
+
+        create_or_update_dummy_file(record: record, doc_attribute: attachment, filename: filename)
+      end
+    end
+  end
+
+  def self.paperclip_storage
+    PAPERCLIP_STORAGE_OPTIONS[:storage]
+  end
+
+  def self.create_or_update_dummy_file(record:, doc_attribute:, filename:)
+    FileUtils.mkdir_p File.dirname(filename)
+    File.open(filename, 'wb') { |file| file.write(SecureRandom.random_bytes(record.send("#{doc_attribute}_file_size"))) }
+
+    record.send("#{doc_attribute}=", File.open(filename))
+    record.save(validate: false)
+  rescue Paperclip::Error => err
+    LogStuff.warn(
+      module: self.name,
+      action: "#{__method__} for #{record.class.table_name}",
+      error: "#{err.class} - #{err.message}"
+    ) { err.message }
+  end
+
+  # Deleting filesystem/s3 files
+  # doc.document.destroy will delete the doc and its meta data in the doc model (not want we want)
+  # doc.document.clear wil delete the paperclip attachment file, without changing the meta data, unless you save
+  # the model object immediatley afterwards.
+
+  # NOTE: for the Document model the `doc.document.clear` message removes the "folder" (tested locally)
+  # which includes the converted_preview_document object too. However, calling `.clear` on a non-existant
+  # attachment does not raise an error.
+  #
+  def self.clear_dummy_paperclip_files_for(model)
```

Comment:
I have tried out and am able to delete the s3 objects directly using code similar to below. this is likely to be a lot quicker.

```
# delete all objects with reports as a prefix
s3_bucket = S3Bucket.new('does-not-matter-if-on-host')
s3_bucket.list('reports').map { |item| item.delete }
```

The disadvantage of this approach is that there is no simple way to distinguish message attachments from evidence document attachments as their s3 keys are both prefixed with `document/...`.